test/configurationTests.ts

```diff
@@ -448,39 +244,63 @@ describe("Configuration", () => {
             assert.isTrue(fs.existsSync(config.rulesDirectory[0]));
             assert.isTrue(fs.existsSync(config.rulesDirectory[1]));
 
-            const expectedConfig = getEmptyConfig();
-            expectedConfig.rules.set("always-fail", { ruleSeverity: "off" });
-            expectedConfig.rules.set("rule-one", { ruleSeverity: "error" });
-            expectedConfig.rules.set("rule-two", { ruleSeverity: "error" });
-            expectedConfig.rules.set("rule-four", { ruleSeverity: "error" });
+            const expectedBaseRules = new Map([
+                ["always-fail", { ruleSeverity: "off" }],
+            ]);
+            assertDeepMapEquals(config.jsRules, expectedBaseRules);
+            assertDeepMapEquals(config.rules, expectedBaseRules);
 
-            assertConfigEquals(config.jsRules, expectedConfig.rules);
-            assertConfigEquals(config.rules, expectedConfig.rules);
+            const expectedFirstExtendRules = new Map<string, Partial<IOptions>>([
+                ["rule-two", {}],
+                ["rule-four", {}],
+            ]);
+            assertDeepMapEquals(config.extends[0].jsRules, expectedFirstExtendRules);
+            assertDeepMapEquals(config.extends[0].rules, expectedFirstExtendRules);
+
+            const expectedSecondExtendRules = new Map<string, Partial<IOptions>>([
+                ["rule-one", {}],
+                ["rule-two", { ruleSeverity: "off" }],
+            ]);
+            assertDeepMapEquals(config.extends[0].extends[0].jsRules, expectedSecondExtendRules);
+            assertDeepMapEquals(config.extends[0].extends[0].rules, expectedSecondExtendRules);
         });
 
         it("extends with array", () => {
             const config = loadConfigurationFromPath("./test/config/tslint-extends-package-array.json");
 
-            const expectedConfig = getEmptyConfig();
-            expectedConfig.rules.set("always-fail", { ruleSeverity: "off" });
-            expectedConfig.rules.set("no-fail", { ruleSeverity: "error" });
-            expectedConfig.rules.set("rule-one", { ruleSeverity: "error" });
-            expectedConfig.rules.set("rule-two", { ruleSeverity: "error" });
+            const expectedBaseRules = new Map([
```

Comment:
This whole series of tests became much more verbose. Before they loaded files off of disk and asserted on the final set of rules.  However rules are not reduced at this point making assertions on a single config less useful. To avoid loosing any test coverage over how files are loaded and parsed, I converted these to asset on the resulting config object and it's `extends` config objects.